### DISCUSSION TITLE
4.13.0: ovnkube-master: Fatal glibc error: CPU does not support x86-64-v2

### DISCUSSION THREAD

**Discussion by kareiva** [okd-project/okd#1620] [Q&A] [answered]
Dear all,

I'm doing a user-provisioned installation of OKD on several virtual FCOS running on bhyve (FreeBSD/TrueNAS). The pods ovnkube-master- and ovnkube-node- are not able to start due to the missing above requirement and, subsequently, glibc error:

```
$ oc logs -n openshift-ovn-kubernetes ovnkube-master-q6tfb -c ovnkube-master
Fatal glibc error: CPU does not support x86-64-v2
```

Is this system of mine really lacking VT/SSE or any other extensions that are required for the control plane to run? Can it be overridden through the `install-config.yaml`?

master1 lscpu: https://p.lr.lt/?hash=8c0a12b7629519211860caf1117ddc76

```
$ lscpu
Architecture:           x86_64
  CPU op-mode(s):       32-bit, 64-bit
  Address sizes:        48 bits physical, 48 bits virtual
  Byte Order:           Little Endian
CPU(s):                 4
  On-line CPU(s) list:  0-3
Vendor ID:              AuthenticAMD
  Model name:           AMD Opteron(tm) Processor 6174
...
    Flags:              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt p
                        dpe1gb lm 3dnowext 3dnow constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid pni cx16 popcnt hypervisor lahf_lm cmp_legacy e
                        xtapic cr8_legacy abm sse4a misalignsse 3dnowprefetch skinit wdt vmmcall npt lbrv svm_lock nrip_save pausefilter

```

host (freebsd) cpuid: (full: https://p.lr.lt/?hash=6ff959a44b940860f2f17274df9d7da6 )

```
% cpuid | grep -iE '(sse|vir)'
VME    Virtual 8086 Mode Enhancements
SSE    SSE extensions
SSE2   SSE2 extensions
VME    Virtual 8086 Mode Enhancements
SVM    Secure Virtual Machine
SSE4A  EXTRQ, INSERTQ, MOVNTSS, and MOVNTSD support
MASSE  Misaligned SSE mode
```

Installer version:
`openshift-install-linux-4.13.0-0.okd-2023-05-22-052007.tar.gz`

Installing control on:
`fedora-coreos-38.20230430.3.1-live.x86_64.iso`

Any help would be greatly appreciated.

**Comment by vrutkovs** [comment, marked as answer]
OVN in 4.13 is based on UBI9 (others are mostly at UBI8), which defines x86-64-2 arch level. See [this article](https://developers.redhat.com/blog/2021/01/05/building-red-hat-enterprise-linux-9-for-the-x86-64-v2-microarchitecture-level#background_of_the_x86_64_microarchitecture_levels) and [this StackOverflow answer](https://unix.stackexchange.com/questions/631217/how-do-i-check-if-my-cpu-supports-x86-64-v2/631218#631218) for more information

**Reply by kareiva** [reply]
Thanks, I was suspecting this and I am now redeploying the cluster with `OpenShiftSDN` to see if it can run on this arch level. 

Very useful sources of information - thanks for sharing.
